include/rmm/device_scalar.hpp

```diff
@@ -82,6 +82,25 @@ class device_scalar {
     set_value(initial_value, stream);
   }
 
+  /**
+   * @brief Construct a new `device_scalar` by deep copying the contents of
+   * another `device_scalar`, using the specified stream and memory
+   * resource.
+   *
+   * @throws rmm::bad_alloc If creating the new allocation fails.
+   * @throws rmm::cuda_error if copying from `other` fails.
+   *
+   * @param other The `device_scalar` whose contents will be copied
+   * @param stream The stream to use for the allocation and copy
+   * @param mr The resource to use for allocating the new `device_scalar`
+   */
+  device_scalar(device_scalar const &other,
+                cudaStream_t stream                 = 0,
+                rmm::mr::device_memory_resource *mr = rmm::mr::get_current_device_resource())
```

Comment:
This changes behavior. The original copy ctor would use the resource from `other`.

Reply:
Not exactly. The original copy ctor calls `device_buffer`'s copy ctor, which defaults to the null stream: https://github.com/rapidsai/rmm/blob/branch-0.17/include/rmm/device_buffer.hpp#L144